build_projects/shared-build-targets-utils/VersionRepoUpdater.cs

```diff
@@ -0,0 +1,126 @@
+
+        private HttpClient _client = new HttpClient();
+        private string _gitHubUser;
+        private string _gitHubEmail;
+        private string _versionsRepoOwner;
+        private string _versionsRepo;
+
+        public VersionRepoUpdater(
+            string gitHubUser = null, 
+            string gitHubEmail = null, 
+            string versionRepoOwner = null, 
+            string versionsRepo = null, 
+            string gitHubAuthToken = null)
+        {
+            _gitHubUser = gitHubUser ?? "dotnet-bot";
+            _gitHubEmail = gitHubEmail ?? "dotnet-bot@microsoft.com";
+            _versionsRepoOwner = versionRepoOwner ?? "dotnet";
+            _versionsRepo = versionsRepo ?? "versions";
+
+            gitHubAuthToken = gitHubAuthToken ?? Environment.GetEnvironmentVariable("GITHUB_PASSWORD");
+
+            if (string.IsNullOrEmpty(gitHubAuthToken))
+            {
+                throw new ArgumentException("A GitHub auth token is required and wasn't provided. Set 'GITHUB_PASSWORD' environment variable.", nameof(gitHubAuthToken));
+            }
+
+            _client.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
+            _client.DefaultRequestHeaders.Add("Authorization", $"token {gitHubAuthToken}");
+            _client.DefaultRequestHeaders.Add("User-Agent", _gitHubUser);
+        }
+
+        public async Task UpdatePublishedVersions(string nupkgFilePath, string versionsRepoPath)
+        {
+            List<Tuple<string, string>> publishedPackages = GetPackageInfo(nupkgFilePath);
+
+            string packageInfoFileContent = string.Join(
+                Environment.NewLine,
+                publishedPackages
+                    .OrderBy(t => t.Item1)
+                    .Select(t => $"{t.Item1} {t.Item2}"));
+
+            string firstVersionWithPrerelease = publishedPackages
+                .FirstOrDefault(t => t.Item2.Contains('-'))
+                ?.Item2;
+
+            string prereleaseVersion = null;
+            if (!string.IsNullOrEmpty(firstVersionWithPrerelease))
+            {
+                prereleaseVersion = firstVersionWithPrerelease.Substring(firstVersionWithPrerelease.IndexOf('-') + 1);
+            }
+
+            string packageInfoFilePath = $"{versionsRepoPath}_Packages.txt";
+            string message = $"Adding package info to {packageInfoFilePath} for {prereleaseVersion}";
+
+            await UpdateGitHubFile(packageInfoFilePath, packageInfoFileContent, message);
+        }
+
+        private static List<Tuple<string, string>> GetPackageInfo(string nupkgFilePath)
+        {
+            List<Tuple<string, string>> packages = new List<Tuple<string, string>>();
+
+            foreach (string filePath in Directory.GetFiles(nupkgFilePath, "*.nupkg"))
+            {
+                Match match = s_nugetFileRegex.Match(Path.GetFileName(filePath));
```

Comment:
I haven't tried. I was just porting the power shell script. What is the concern - that the regex would be flaky or the format would change?